src/OpenStudy/vector_map.jl

```diff
@@ -0,0 +1,315 @@
+# see main definition
+function mapped_vector(
+    data::Data,
+    col::String,
+    name::String,
+    ::Type{T},
+    dim1::String="",
+    dim2::String="";
+    ignore::Bool=false,
+    map_key = col, # reference for PSRMap pointer, if empty use class name
+    filters = String[], # for calling just within a subset instead of the full call
+) where T #<: Union{Float64, Int32}
+
+    raw = _raw(data)
+
+    n = max_elements(data, col)
+    if n == 0
+        return T[]
+    end
+
+    attr_struct = get_attribute_struct(data, col, name)
+
+    _check_type(attr_struct, T, col, name)
+    _check_vector(attr_struct, col, name)
+
+    # validate dimensions
+    dim = _check_dim(attr_struct, col, name, dim1, dim2)
+
+    dim1_val = _add_get_dim_val(data, dim1)
+    dim2_val = _add_get_dim_val(data, dim2)
+
+    total_dim = dim1_val + dim2_val
+    if total_dim != dim
+        error("Dimension mismatch, data structure should have $(total_dim) but has $dim in the data file")
+    end
+
+    index = attr_struct.index
+    stage = data.controller_stage
+
+    cache = _get_cache(data, T)
+
+    col_cache = get!(cache, col, Dict{String, VectorCache{T}}())
+
+    if haskey(col_cache, name)
+        error("Attribute $name was already mapped.")
+    end
+
+    out = T[_default_value(T) for _ in 1:n] #zeros(T, n)
```

Comment:
```julia
out = fill(_default_value(T), n)
```

Reply:
this is dangerous because `Dates.Date` is mutable. So all entries will be the same and then if one modifies one, will modify all...